src/cdk/platform/features/shadow-dom.ts

```diff
@@ -9,11 +9,24 @@
 let shadowDomIsSupported: boolean;
 
 /** Checks whether the user's browser support Shadow DOM. */
-export function _supportsShadowDom(): boolean {
+export function supportsShadowDom(): boolean {
   if (shadowDomIsSupported == null) {
     const head = typeof document !== 'undefined' ? document.head : null;
     shadowDomIsSupported = !!(head && ((head as any).createShadowRoot || head.attachShadow));
   }
 
   return shadowDomIsSupported;
 }
+
+/** Gets the shadow root of an element, if supported and the element is inside the Shadow DOM. */
+export function getShadowRoot(element: HTMLElement): Node | null {
```

Comment:
Is cdk/platform the right place for this? I'm not sure we have a package that really makes sense for it currently. I would think something like cdk/dom would be the right place

Reply:
I personally could see it live in something like `cdk/platform/dom`.

Reply:
I'm extremely hesitant to get into the business of general DOM helper APIs. I think it's a slippery slope in terms of adding more, and they tend to go out of date over the years. I'm okay with this living here if we keep it as a private API.

Reply:
It's not really private if we export it at the top level though. Also we're already in the business of detecting other DOM APIs like supported input types and scroll behavior so I don't see how detecting if the Shadow DOM is supported is any different.

Reply:
We have a `cdk/testing/private`, maybe we could make `cdk/platform/private` (I'm less concerned about it being in this package if its private)

Reply:
I'd be fine with putting it under `private`. The main concern I'm trying to address with these changes is to avoid the code duplication.